ansible/configs/multi-region-tower/files/cloud_providers/default.j2

```diff
@@ -0,0 +1,369 @@
+        Ref: VpcRouteTable
+
+  VpcGA:
+    Type: "AWS::EC2::VPCGatewayAttachment"
+    Properties:
+      InternetGatewayId:
+        Ref: VpcInternetGateway
+      VpcId:
+        Ref: Vpc
+
+  PublicSubnet:
+    Type: "AWS::EC2::Subnet"
+    DependsOn:
+      - Vpc
+    Properties:
+    {% if aws_availability_zone is defined %}
+      AvailabilityZone: {{ aws_availability_zone }}
+    {% endif %}
+
+      CidrBlock: "{{ aws_public_subnet_cidr }}"
+      Tags:
+        - Key: Name
+          Value: "{{project_tag}}"
+        - Key: Hostlication
+          Value:
+            Ref: "AWS::StackId"
+      MapPublicIpOnLaunch: true
+      VpcId:
+        Ref: Vpc
+
+  PublicSubnetRTA:
+    Type: "AWS::EC2::SubnetRouteTableAssociation"
+    Properties:
+      RouteTableId:
+        Ref: VpcRouteTable
+      SubnetId:
+        Ref: PublicSubnet
+
+{% for security_group in security_groups|list + default_security_groups|list %}
+  {{security_group['name']}}:
+    Type: "AWS::EC2::SecurityGroup"
+    Properties:
+      GroupDescription: Host
+      VpcId:
+        Ref: Vpc
+      Tags:
+        - Key: Name
+          Value: "{{security_group['name']}}"
+{% endfor %}
+
+{% for security_group in default_security_groups|list + security_groups|list %}
+{% for rule in security_group.rules %}
+  {{security_group['name']}}{{rule['name']}}:
+    Type: "AWS::EC2::SecurityGroup{{rule['rule_type']}}"
+    Properties:
+     GroupId:
+       Fn::GetAtt:
+         - "{{security_group['name']}}"
+         - GroupId
+     IpProtocol: {{rule['protocol']}}
+     FromPort: {{rule['from_port']}}
+     ToPort: {{rule['to_port']}}
+  {% if rule['cidr'] is defined %}
+     CidrIp: "{{rule['cidr']}}"
+  {% endif  %}
+
+  {% if rule['from_group'] is defined %}
+     SourceSecurityGroupId:
+       Fn::GetAtt:
+        - "{{rule['from_group']}}"
+        - GroupId
+  {% endif  %}
+{% endfor %}
+{% endfor %}
+
+  DnsZonePrivate:
+    Type: "AWS::Route53::HostedZone"
+    Properties:
+      Name: "{{ aws_dns_zone_private }}"
+      VPCs:
+        - VPCId:
+            Ref: Vpc
+          VPCRegion:
+            Ref: "AWS::Region"
+      HostedZoneConfig:
+        Comment: "{{ aws_comment }}"
+
+  {% if secondary_stack is not defined %}
+  DnsZonePublic:
+    Type: "AWS::Route53::HostedZone"
+    Properties:
+      Name: "{{ aws_dns_zone_public }}"
+      HostedZoneConfig:
+        Comment: "{{ aws_comment }}"
+
+  DnsPublicDelegation:
+    Type: "AWS::Route53::RecordSetGroup"
+    DependsOn:
+      - DnsZonePublic
+    Properties:
+    {% if HostedZoneId is defined %}
+      HostedZoneId: "{{ HostedZoneId }}"
+    {% else %}
+      HostedZoneName: "{{ aws_dns_zone_root }}"
+    {% endif %}
+      RecordSets:
+        - Name: "{{ aws_dns_zone_public }}"
+          Type: NS
+          TTL: {{ aws_dns_ttl_public }}
+          ResourceRecords:
+            "Fn::GetAtt":
+              - DnsZonePublic
+              - NameServers
+    {% endif %}
+
+{% for instance in instances %}
```

Comment:
We must provide a mean to user to specify the region per-instance in the `instances` variable.
Then filter here.
This is how i did it:

```jinja2
{% for instance in instances
  if ('regions' in instance
       and aws_region in instance.regions)
     or 'regions' not in instance
  %}
```

But we can use the `Worker_region` tag too.